src/main/java/org/zalando/nakadi/controller/EventStreamController.java

```diff
@@ -98,6 +101,8 @@ public StreamingResponseBody streamEvents(
                 final EventType eventType = eventTypeRepository.findByName(eventTypeName);
                 final String topic = eventType.getTopic();
 
+                client.authorize(eventType.getReadScopes());
```

Comment:
Now I realize yet another problem. This authorisation logic is specific to streaming and now it's coupled with the client. Maybe my suggestion wasn't the best. But anyway, I think it's good as is. When new authorisation rules are added, we can always refactor. Let's be pragmatic.